int32 id
float32 pos
float32 speed
float32 current